l2gersync/evm_downloader_sovereign.go

```diff
@@ -80,16 +83,24 @@ func (d *downloaderSovereign) Download(ctx context.Context, fromBlock uint64, do
 		case <-ctx.Done():
 			log.Debug("aborting the l2GERSync downloader...")
 			close(downloadedCh)
-
 			return
 		default:
 		}
 
-		// Wait for new blocks before processing
-		fromBlock = d.WaitForNewBlocks(ctx, fromBlock)
-		for _, block := range d.GetEventsByBlockRange(ctx, fromBlock, fromBlock) {
+		// Wait for new blocks and get current head
+		latestBlock := d.WaitForNewBlocks(ctx, fromBlock)
+		toBlock := fromBlock + d.syncBlockChunkSize - 1
+		if toBlock > latestBlock {
+			toBlock = latestBlock
+		}
```

Comment:
This is a recommendation from the IDE, to simplify this part, like this
```suggestion
		toBlock := min(fromBlock + d.syncBlockChunkSize - 1, latestBlock)
```

Reply:
nice one